src/storage/core.rs

```diff
@@ -315,6 +315,46 @@ impl<K: Key> Storage<K> {
             .with_context(|| "no results in closed blobs")
     }
 
+    /// Mark as deleted entries with matching key
+    /// # Errors
+    /// Fails after any disk IO errors.
+    pub async fn mark_all_as_deleted(&self, key: impl AsRef<K>) -> Result<u64> {
+        let mut total_count = 0u64;
+        let mut safe = self.inner.safe.write().await;

```

Comment:
You should split this function into `mark_all_as_deleted_active` and `mark_all_as_deleted_closed`, so it's intentions are easier to find out

Reply:
Done